### DISCUSSION TITLE
ERROR WITH WEBSITE

### DISCUSSION THREAD

**Discussion by Artyomcha** [supabase/supabase#27996] [Questions]
I am trying to check for the already existing data in Superbas table the similarity with the input data, and it constantly gives me some problems in terminal console, so what can I do to Mae changes in such way so I can constantly add new info to table and compare it through the embeddings technologies, so can you help me have to figure out this error?:

code:

    embedding = await create_embedding(input_text)
    match = await find_nearest_match(embedding)
    if match:
        await get_chat_completion(match, input_text)
    else:
        print("No matching document found.")

# Create an embedding vector representing the input text
async def create_embedding(input_text):
    response = client.embeddings.create(
        model="text-embedding-3-small",
        input=input_text
    )
    return response.data[0].embedding

# Query Supabase and return a semantically matching text chunk
async def find_nearest_match(embedding):
    response = supabase.rpc('documents', {
        'query_embedding': embedding,
        'match_threshold': 0,
        'match_count': 1
    }).execute()
    if response.data and len(response.data) > 0:
        return response.data[0]['content']
    else:
        return None

# Use OpenAI to make the response conversational
async def get_chat_completion(text, query):
    chat_messages = [{
        'role': 'system',
        'content': (
            "You are an enthusiastic podcast expert who loves recommending podcasts to people. "
            "You will be given two pieces of information - some context about podcasts episodes "
            "and a question. Your main job is to formulate a short answer to the question using the provided context. "
            "If you are unsure and cannot find the answer in the context, say, 'Sorry, I don't know the answer.' "
            "Please do not make up the answer."
        )
    }, {
        'role': 'user',
        'content': f"Context: {text} Question: {query}"
    }]

    response = client.chat.completions.create(
        model='gpt-4o',
        messages=chat_messages,
        temperature=0.5,
        frequency_penalty=0.5
    )

    print(response['choices'][0]['message']['content'])

if __name__ == "__main__":
    asyncio.run(main(query))```

error:
 ```postgrest.exceptions.APIError: {'code': 'PGRST202', 'details': 'Searched for the function public.documents with parameters match_count, match_threshold, query_embedding or with a single unnamed json/jsonb parameter, but no matches were found in the schema cache.', 'hint': 'Perhaps you meant to call the function public.match_documents', 'message': 'Could not find the function public.documents(match_count, match_threshold, query_embedding) in the schema cache'}``

**Comment by GaryAustin1** [comment]
It is suggesting to call a function match_documents not documents.

https://supabase.com/docs/guides/ai/vector-columns#querying-a-vector--embedding
Shows the function the rpc call is supposed to use.  Did you generate this function?

**Reply by Artyomcha** [reply]
yes, -- Create a function to search for documents
create or replace function match_documents (
  query_embedding vector(1536),
  match_threshold float,
  match_count int
)
returns table (
  id bigint,
  content text,
  similarity float
)
language sql stable
as $$
  select
    documents.id,
    documents.content,
    1 - (documents.embedding <=> query_embedding) as similarity
  from documents
  where 1 - (documents.embedding <=> query_embedding) > match_threshold
  order by similarity desc
  limit match_count;
$$;

**Reply by GaryAustin1** [reply]
![image](https://github.com/user-attachments/assets/c2c39b52-ebb0-46d1-8e02-b3e939e74d29)

You are calling documents function not match_documents function.

**Reply by Artyomcha** [reply]
async def main(input_text):
    embedding = await create_embedding(input_text)
    print(f"Generated Embedding: {embedding}")  # Debugging: Print the generated embedding
    match = await find_nearest_match(embedding)
    if match:
        await get_chat_completion(match, input_text)
    else:
        print("No matching document found.")

# Create an embedding vector representing the input text
async def create_embedding(input_text):
    response = client.embeddings.create(
        model="text-embedding-ada-002",
        input=input_text
    )
    return response.data[0].embedding

# Query Supabase and return a semantically matching text chunk
async def find_nearest_match(embedding):
    # Ensure the embedding is in the correct format
    embedding = [float(value) for value in embedding]
    
    response = supabase.rpc('match_documents', {
        'query_embedding': embedding,
        'match_threshold': 0.01,
        'match_count': 1
    }).execute()
    print(f"Supabase Response: {response.data}")  # Debugging: Print the response from Supabase
    if response.data and len(response.data) > 0:
        return response.data[0]['content']
    else:
        return None

# Use OpenAI to make the response conversational
async def get_chat_completion(text, query):
    chat_messages = [{
        'role': 'system',
        'content': (
            "You are an enthusiastic podcast expert who loves recommending podcasts to people. "
            "You will be given two pieces of information - some context about podcasts episodes "
            "and a question. Your main job is to formulate a short answer to the question using the provided context. "
            "If you are unsure and cannot find the answer in the context, say, 'Sorry, I don't know the answer.' "
            "Please do not make up the answer."
        )
    }, {
        'role': 'user',
        'content': f"Context: {text} Question: {query}"
    }]


    response = client.chat.completions.create(
        model='gpt-4o',
        messages=chat_messages,
        temperature=0.5,
        frequency_penalty=0.5
    )

    print(response['choices'][0]['message']['content'])

# Example usage
if __name__ == "__main__":
    asyncio.run(main(query))

**Reply by Artyomcha** [reply]
not working as well
same error
